phoenix-pherf/src/main/java/org/apache/phoenix/pherf/Pherf.java

```diff
@@ -146,49 +173,77 @@ public void run() throws Exception {
                 }
                 return;
             }
-            workloadExec = (scenarioFile == null)
-                    ? new WorkloadExecutor(properties,
-                    new XMLConfigParser(PherfConstants.DEFAULT_FILE_PATTERN),
-                    monitor)
-                    : new WorkloadExecutor(properties,
-                    new XMLConfigParser(scenarioFile),
-                    monitor);
+            XMLConfigParser parser = new XMLConfigParser(scenarioFile);
 
             // Drop tables with PHERF schema and regex comparison
             if (null != dropPherfTablesRegEx) {
-                logger.info("\nDropping existing table with PHERF namename and "
-                        + dropPherfTablesRegEx + " regex expression.");
-                new PhoenixUtil().deleteTables(dropPherfTablesRegEx);
+                logger.info(
+                        "\nDropping existing table with PHERF namename and " + dropPherfTablesRegEx
+                                + " regex expression.");
+                phoenixUtil.deleteTables(dropPherfTablesRegEx);
             }
 
-            // Schema and Data Load
-            if (loadData) {
+            if (monitor) {
+                MonitorManager
+                        monitorManager =
+                        new MonitorManager(Integer.parseInt(
+                                properties.getProperty("pherf.default.monitorFrequency")));
+                workloadExecutor.add(monitorManager);
```

Comment:
Should we wait for monitor job to be executed or kicked off by doing a workloadExecutor.get() or workloadExecutor.get(monitorManager)? Or is the intent to rely on subsequent workloadExecutor.get() calls to have them wait on MonitorManager.